src/Events/Appointments/AppointmentsFilterEvent.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * This file is part of OpenEMR.
+ *
+ * @link https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Events\Appointments;
+
+use OpenEMR\Services\UserService;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event object for creating custom appointment filters, affecting the queries for patient tracker
+ *
+ * @package OpenEMR\Events
+ * @subpackage Appointments
+ * @author Ken Chapple <ken@mi-squared.com>
+ * @copyright Copyright (c) 2019 Ken Chapple <ken@mi-squared.com>
+ */
+class AppointmentsFilterEvent extends Event
+{
+    /**
+     * The customFilter event occurs in the library/appointments.inc.php file in the fetchEvents()
+     * function, which fetches an array of all appointments. Setting this object's customWhereFilter
+     * can filter appointments that show up.
+     */
+    const EVENT_HANDLE = 'appointments.customFilter';
+
+    /**
+     * @var null|UserService
+     *
+     */
+    private $userService = null;
+
+    /**
+     * @var string
+     *
+     * This is the custom filter that can add to the appointment fetching query
+     */
+    private $customWhereFilter= '1';
+
+    /**
+     *
+     */
+    public function __construct(UserService $userService)
```

Comment:
I usually try to pass in object trying to stick to principles of encapsulation and object composition. I live in a dream world where some day maybe we'll get to unit testing modules :)  I get what you're saying though, and it can be instantiated anywhere so I can remove it if it bugs you.